driver-core/src/main/com/mongodb/operation/AsyncQueryBatchCursor.java

```diff
@@ -73,6 +73,8 @@
     private volatile BsonDocument postBatchResumeToken;
     private volatile BsonTimestamp operationTime;
     private volatile boolean firstBatchEmpty;
+    private final AtomicBoolean noOperationInProgress = new AtomicBoolean(true);
```

Comment:
Nit: For consistency with `isClosed`, prefix these booleans with `is`, e.g. `isOperationInProgress` and `isClosePending`.

Reply:
Done